### ISSUE TITLE
CMake policy violated (CMP0026)

### ISSUE THREAD

**Issue by hmeine** [ukoethe/vigra#243]
CMake 3.0.2 produces a lot of warnings, referring to this issue:

> CMP0026
> -------
> 
> Disallow use of the LOCATION target property.
> 
> CMake 2.8.12 and lower allowed reading the LOCATION target
> property (and configuration-specific variants) to
> determine the eventual location of build targets.  This relies on the
> assumption that all necessary information is available at
> configure-time to determine the final location and filename of the
> target.  However, this property is not fully determined until later at
> generate-time.  At generate time, the $\<TARGET_FILE\> generator
> expression can be used to determine the eventual LOCATION of a target
> output.
> 
> Code which reads the LOCATION target property can be ported to use the
> $\<TARGET_FILE\> generator expression together with the file(GENERATE)
> subcommand to generate a file containing the target location.
> 
> The OLD behavior for this policy is to allow reading the LOCATION
> properties from build-targets.  The NEW behavior for this policy is to
> not to allow reading the LOCATION properties from build-targets.
> 
> This policy was introduced in CMake version 3.0.  CMake version
> 3.0.2 warns when the policy is not set and uses OLD behavior.  Use
> the cmake_policy command to set it to OLD or NEW explicitly.

**Comment by ukoethe** [comment]
Yes, I know. But the explanations in the cmake documentation (partly cited above) are quite bad, so that I'm unable to find out how to fix this. Suggestions?

**Comment by hmeine** [comment]
I am not familiar with CMake at all; I would rather hope for someone else to chime in.

**Comment by stuarteberg** [comment]
I just opened a PR to hide this warning.  On mac, one more cmake policy warning remains, but I've opened a separate issue for that (#245).

**Comment by ukoethe** [comment]
Fixed in 78ed77635754837410c4743c534b2cf4d0f2d3a6

**Comment by ArchangeGabriel** [comment]
You’re running into troubles soon though now:
```
CMake Deprecation Warning at config/VigraAddTest.cmake:66 (cmake_policy):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.
```
They were 56 such messages during build. So a proper fixing will soon be required.
